### PR TITLE
ZCL_ABAPGIT_SERVICES_GIT: Add new method checkout_commit to let the user check out commits

### PR DESCRIPTION
**Pull request by mariusraht2**
Related to #3040

Adds a new method checkout_commit to ZCL_ABAPGIT_SERVICES_GIT that makes it possible to select commits from a popup and set the repo commit status after the selection. Added caller in GUI_ROUTER and some pretty printer changes too. Had to extend the hash column to be able to show the 7 first instead of just the 6 first hash characters.

### CODE REVIEW

**Comment by larshp** [comment]
method `checkout_commit` is not called from anywhere yet? suggest adding a bit more code

the method is new, so it wont break anything

File: src/ui/zcl_abapgit_services_git.clas.abap [hunk 1]
```diff
@@ -93,6 +98,38 @@ ENDCLASS.
...
+        iv_url         = lo_repo->get_url( )
+      IMPORTING
+        et_value_tab   = lt_value_tab
+        et_commits     = lt_commits ).
+
+    ls_selected_commit = checkout_commit_build_popup( it_commits   = lt_commits
+                                                      it_value_tab = lt_value_tab ).
+
+    lo_repo->set_sha1( ls_selected_commit-sha1 ).
+    COMMIT WORK AND WAIT.
```
Comment (larshp) [inline src/ui/zcl_abapgit_services_git.clas.abap:128]: How will the refreshing of the repository work? this just sets a sha1 value, ie. the repository might be in an inconsistent state?

**Review by larshp — CHANGES_REQUESTED** [review]
see comment above

File: src/ui/zcl_abapgit_services_git.clas.abap (hunk 1 above)
Comment (mariusraht2) [reply src/ui/zcl_abapgit_services_git.clas.abap:128]: In set_sha1 calling reset_remote( ) was missing. Thanks 👍

Comment (mariusraht2) [reply src/ui/zcl_abapgit_services_git.clas.abap:128]: Maybe we should split this PR? It gets a little bit confusing, right?

**Review by larshp — APPROVED** [review]

**Pull request merged**